dns/asyncresolver.py

```diff
@@ -394,3 +394,76 @@ async def zone_for_name(
             name = name.parent()
         except dns.name.NoParent:  # pragma: no cover
             raise NoRootSOA
+
+
+async def make_resolver_at(
+    where: Union[dns.name.Name, str],
+    port: int = 53,
+    family: int = socket.AF_UNSPEC,
+) -> Resolver:
+    """Make a stub resolver using the specified destination as the full resolver.
+
+    *where*, a ``dns.name.Name`` or ``str`` the domain name or IP address of the
+    full resolver.
+
+    *port*, an ``int``, the port to use.  If not specified, the default is 53.
+
+    *family*, an ``int``, the address family to use.  This parameter is used if
+    *where* is not an address.  The default is ``socket.AF_UNSPEC`` in which case
+    the first address returned by ``resolve_name()`` will be used, otherwise the
+    first address of the specified family will be used.
+
+    Returns a ``dns.resolver.Resolver`` or raises an exception.
+    """
+    if isinstance(where, str) and dns.inet.is_address(where):
+        address = where
+    else:
+        answers = await resolve_name(where, family)
```

Comment:
It looks like this will use call `resolve_name` with the default resolver?  Should there be a way to use an alternate resolver for that resolution?

Reply:
I was trying to keep it super simple, but adding a resolver parameter is not too painful.